File: python/03_008.py
def get_permutations(string):
    # Base case
    if len(string) <= 1:
        return [string]
    
    # Recursive case
    perms = []
    for i in range(len(string)):
        char = string[i]
        remaining = string[:i] + string[i+1:]
        
        for p in get_permutations(remaining):
            perms.append(char + p)
            
    return perms

# Example usage
result = get_permutations("abc")
print(result)  # ['abc', 'acb', 'bac', 'bca', 'cab', 'cba']

File: java/03_008.java
import java.util.ArrayList;
import java.util.List;

public class StringPermutations {
  public static List<String> getPermutations(String string) {
     // Base case
     List<String> result = new ArrayList<>();
     if (string.length() <= 1) {
        result.add(string);
        return result;
     }
     
     // Recursive case
     for (int i = 0; i < string.length(); i++) {
        char currentChar = string.charAt(i);
        String remaining = string.substring(0, i) + 
           string.substring(i + 1);
        
        for (String perm : getPermutations(remaining)) {
           result.add(currentChar + perm);
        }
     }
     
     return result;
  }
  
  public static void main(String[] args) {
     List<String> result = getPermutations("abc");
     System.out.println(result);  // [abc, acb, bac, bca, cab, cba]
  }
}